bin/sg-otio.py

```diff
@@ -0,0 +1,314 @@
+        required=True,
+        help="The SG Entity type to link the Cut to. It can be a Project, a Cut, or any Entity a Cut can be linked to, "
+             "e.g. a Sequence, a Reel."
+    )
+    write_parser.add_argument(
+        "--entity-id", "-i",
+        required=True,
+        type=int,
+        help="The SG Entity id to link the Cut to."
+    )
+    write_parser.add_argument(
+        "--movie", "-m",
+        required=False,
+        help="An optional movie of the timeline so that it can be published as a Version and used to create Versions"
+             "for each clip"
+    )
+
+    # Compare parameters
+    compare_parser.add_argument(
+        "--file", "-f",
+        required=True,
+        help="The file to read the Timeline from (can be OTIO, EDL, etc)."
+    )
+    compare_parser.add_argument(
+        "--cut-id", "-i",
+        required=True,
+        help="The ID of the SG Cut to read."
+    )
+
+    args = parser.parse_args()
+
+    # Check that we can log in to SG
+    if not (args.login and args.password):
+        if not (args.script_name and args.api_key):
+            if not args.session_token:
+                raise ValueError(
+                    "You must provide either a login, password, script name and API key or a session token."
+                )
+    if args.verbose:
+        logging.getLogger().setLevel(logging.DEBUG)
+    if args.command == "read":
+        read_from_sg(args)
+    elif args.command == "write":
+        write_to_sg(args)
+    elif args.command == "compare":
+        compare_to_sg(args)
+
+
+def read_from_sg(args):
+    """
+    Reads information from a Cut in ShotGrid and print it
+    to stdout, or write it to a file.
+
+    If printed to stdout, the output is in OTIO format.
+    If written to a file, the output depends on the extension of the file
+    or the adapter chosen.
+
+    The retrieved SG entities are stored in the items metadata.
+
+    :param args: The command line arguments.
+    """
+    url = get_read_url(
+        sg_site_url=args.sg_site_url,
+        cut_id=args.cut_id,
+        session_token=_get_session_token(args)
+    )
+    timeline = otio.adapters.read_from_file(
+        url,
+        adapter_name="ShotGrid",
+    )
+    if not args.file:
+        logger.info(otio.adapters.write_to_string(timeline))
+    else:
+        otio.adapters.write_to_file(timeline, args.file, adapter_name=args.adapter)
+
+
+def write_to_sg(args):
+    """
+    Write an input file to SG as a Cut.
+
+    The input file can be any format supported by OTIO.
+    If no adapter is provided, the default adapter for the file extension is used.
+
+    :param args: The command line arguments.
+    """
+    session_token = _get_session_token(args)
+    url = get_write_url(
+        sg_site_url=args.sg_site_url,
+        entity_type=args.entity_type,
+        entity_id=args.entity_id,
+        session_token=session_token
+    )
+    timeline = otio.adapters.read_from_file(
```

Comment:
I think there's something missing here for EDLs. If they're not 24fps, normally you need to provide the frame rate. I think we need to add an argument and add it here if provided. Maybe we should create a ticket for this?

Reply:
Ticket here: https://gpltech.shotgunstudio.com/detail/Ticket/9322